<?php
namespace App\Exceptions; use App\Library\Helper; use App\Library\Response; use Exception; use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler; use Illuminate\Support\Facades\Log; class Handler extends ExceptionHandler { protected $dontReport = array(); protected $dontFlash = array('password', 'password_confirmation'); public function report(Exception $sp0ee84b) { parent::report($sp0ee84b); } private function msg($spa27895, $sp478b36, $spf66c3d) { return $spa27895->isXmlHttpRequest() ? response()->json(array('message' => $sp478b36), $spf66c3d) : response()->view('errors._', array('code' => $spf66c3d, 'message' => $sp478b36), $spf66c3d); } public function render($spa27895, Exception $sp45f5f1) { if ($sp45f5f1 instanceof \Illuminate\Database\Eloquent\ModelNotFoundException) { return $this->msg($spa27895, '记录未找到', 404); } elseif ($sp45f5f1 instanceof \Illuminate\Auth\AuthenticationException) { return $this->msg($spa27895, '您需要登录您的账户再进行此操作', 401); } elseif ($sp45f5f1 instanceof \Illuminate\Auth\Access\AuthorizationException) { return $this->msg($spa27895, '未授权的操作', 403); } elseif ($sp45f5f1 instanceof \Illuminate\Validation\ValidationException) { return parent::render($spa27895, $sp45f5f1); } elseif ($sp45f5f1 instanceof \Illuminate\Session\TokenMismatchException) { return $this->msg($spa27895, '请刷新页面后重试', 403); } elseif ($sp45f5f1 instanceof \Symfony\Component\HttpKernel\Exception\NotFoundHttpException) { return $this->msg($spa27895, '页面未找到', 404); } elseif ($sp45f5f1 instanceof \Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException) { return $this->msg($spa27895, '请求方法不允许', 405); } elseif ($sp45f5f1 instanceof \Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException) { Log::error('Caught a ServiceUnavailableHttpException', array('Exception' => $sp45f5f1)); return $this->msg($spa27895, '当前服务不可用，请稍后重试', 503); } elseif ($sp45f5f1 instanceof \Symfony\Component\HttpKernel\Exception\HttpException) { $sp1b2df3 = $sp45f5f1->getStatusCode(); switch ($sp1b2df3) { case 429: return $this->msg($spa27895, '您的请求过于频繁，请稍后重试', 429); break; default: Log::error('Caught a UnknownHttpException', array('Exception' => $sp45f5f1)); return $this->msg($spa27895, '当前服务不可用，请稍后重试', 500); } } Log::error('Uncaught error', array('exception' => $sp45f5f1, 'method' => $spa27895->method(), 'url' => $spa27895->fullUrl(), 'data' => file_get_contents('php://input'), 'ip' => Helper::getIP(), 'client_ip' => $spa27895->getClientIp(), 'headers' => $spa27895->header())); if (config('app.debug')) { return parent::render($spa27895, $sp45f5f1); } else { return $this->msg($spa27895, '未知错误，请联系客服', 500); } } }